verification/verify.go

```diff
@@ -149,7 +151,15 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 	return nil, err
 }
 
+func IsFatal(err error) bool {
+	_, fatal := err.(Fatal)
+	return fatal
+}
+
 func IsRetryable(err error) bool {
 	_, retryable := err.(Retryable)
-	return retryable
+	if retryable {
+		return true
+	}
+	return IsFatal(err)
```

Comment:
How about `return retryable || IsFatal(err)`

Reply:
nice catch! :-) I applied and squashed it into bb32d4cd02a40246f6ff1818d56507a6884531ef